addons/point_of_sale/models/pos_session.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class PosSession(models.Model):
+    _name = 'pos.session'
+    _order = 'id desc'
+
+    POS_SESSION_STATE = [
+        ('opening_control', 'Opening Control'),  # Signal open
+        ('opened', 'In Progress'),               # Signal closing
+        ('closing_control', 'Closing Control'),  # Signal close
+        ('closed', 'Closed & Posted'),
+    ]
+
+    def _confirm_orders(self):
+        PosOrder = self.env['pos.order']
+        for session in self:
+            company_id = session.config_id.journal_id.company_id.id
+            local_context = dict(self.env.context, force_company=company_id)
+            orders = session.order_ids.filtered(lambda order: order.state == 'paid')
+            move = PosOrder.with_context(local_context)._create_account_move(session.start_at, session.name, session.config_id.journal_id.id, company_id)
```

Comment:
This is wrong, the initial code was `pos_order_obj._create_account_move(cr, uid, session.start_at, session.name, session.config_id.journal_id.id, company_id, context=context)` **not** `local_context`